frontend/src/scenes/sessions/SessionDetails.tsx

```diff
@@ -1,15 +1,32 @@
-import React, { useState } from 'react'
+import React, { useEffect, useState } from 'react'
 import { Table } from 'antd'
-import { humanFriendlyDiff, humanFriendlyDetailedTime } from '~/lib/utils'
+import { humanFriendlyDiff, humanFriendlyDetailedTime, toParams } from '~/lib/utils'
 import { EventDetails } from 'scenes/events'
 import { Property } from 'lib/components/Property'
 import { eventToName } from 'lib/utils'
-import { EventType } from '~/types'
+import { EventType, SessionType } from '~/types'
+import api from 'lib/api'
+
+export function SessionDetails({ session }: { session: SessionType }): JSX.Element {
+    const [events, setEvents] = useState(session.events || [])
+    const [eventsLoading, setEventsLoading] = useState(session.events === undefined)
 
-export function SessionDetails({ events }: { events: EventType[] }): JSX.Element {
     const [page, setPage] = useState(1)
     const [pageSize, setPageSize] = useState(50)
 
+    async function loadSessionEvents(): Promise<void> {
```

Comment:
i think we're trying to keep most data logic in kea logics if possible. even if this data is scoped here only it'll be more modular if there's an accompanying logic instead of localstate

Reply:
In this case the data was only used locally and the kea logic would have ended up quite a bit more convoluted. 

But moving this out will become neccessary soon as we will want to show highlighted events in session recording. Ok if I punt on this until then?

Reply:
yup just a heads up not strict